### PR TITLE
fix(tests): Fix tests in Linux and macOS

### PR DESCRIPTION
**Pull request by niheaven**
#### Description

1. Skip (Exclude) Windows-only tests and decompress tests
2. Remove old test framework codes
3. Make all tests passed in Linux/macOS

#### Motivation and Context

Only have MacBook Air at home, so fix the tests to maintain Scoop code.

#### How Has This Been Tested?

Windows:

image

Ubuntu:

image

#### Checklist:

- [x] I have read the [Contributing Guide](https://github.com/ScoopInstaller/.github/blob/main/.github/CONTRIBUTING.md).
- [x] I have ensured that I am targeting the `develop` branch.
- [ ] I have updated the documentation accordingly.
- [x] I have updated the tests accordingly.
- [x] I have added an entry in the CHANGELOG.

### CODE REVIEW

File: lib/decompress.ps1 [hunk 1]
```diff
@@ -240,7 +240,7 @@ function Expand-ZipArchive {
         $OriDestinationPath = $DestinationPath
         $DestinationPath = "$DestinationPath\_tmp"
     }
-    Expand-Archive -Path $Path -DestinationPath $DestinationPath -Force
+    Microsoft.PowerShell.Archive\Expand-Archive -Path $Path -DestinationPath $DestinationPath -Force
```
Comment (chawyehsu) [inline lib/decompress.ps1:243]: Move this to a separate PR.

Comment (niheaven) [reply lib/decompress.ps1]: Really don't know why there's people using Pscx v3.

Comment (chawyehsu) [reply lib/decompress.ps1]: Because it's still in beta on PowerShell Gallery. Most people won't install beta software by default you know.

And you never know: https://github.com/ScoopInstaller/Scoop/pull/4638#discussion_r782685435 

`¯\_(ツ)_/¯`

Comment (niheaven) [reply lib/decompress.ps1]: Yes since I don't use it. And, that's added by myself in #3425, sigh...
